### PR TITLE
Modify Stack type interface in preparation for new implementation

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: lib/fizzy/stack.hpp [hunk 1]
```diff
@@ -30,9 +30,13 @@ class Stack : public std::vector<T>
         return res;
     }
 
-    T peek(size_t depth = 0) const noexcept { return (*this)[size() - depth - 1]; }
+    T& operator[](size_t index) noexcept { return std::vector<T>::operator[](size() - index - 1); }
```
Comment (gumb0) [inline lib/fizzy/stack.hpp:33]: What about `const` versions of `operator[]` and `top`?

Comment (chfast) [reply lib/fizzy/stack.hpp:33]: Not needed (Stack is never used as const).

Comment (gumb0) [reply lib/fizzy/stack.hpp:33]: What if it is at some point in some functios, will then `vector`'s version with opposite semantics be called or a confusing compiler message output?

Comment (chfast) [reply lib/fizzy/stack.hpp:33]: This is excerpt from #265 to make the other PR smaller. There `OperandStack` does not extend `std::vector`. Can be merge as part of #265 if that's preferred.

See also #260

Comment (gumb0) [reply lib/fizzy/stack.hpp:33]: But it still doesn't delete `Stack` class.
Can you `=delete` const versions, if this class doesn't allow them?

**Pull request closed without merging**